src/js/components/question-input/question-input.js

```diff
@@ -0,0 +1,175 @@
+import Preview from "./preview";
+import "./question-input.scss";
+
+import { SimpleSelect } from "react-selectize";
+import { buildOptions } from "../../../lib/helpers/questions-helpers";
+
+export const TEXTAREA_PLACEHOLDER = "Enter your line here.";
+export const TYPE_OPTIONS = buildOptions(['Code', 'Text']);
+export const VIEW_OPTIONS = buildOptions(['Normal', 'Italics', 'Bolded']);
+
+export default class QuestionInput extends React.Component {
+	constructor(props) {
+	    super(props);
+	    this.typeOptions = TYPE_OPTIONS;
+	    this.viewOptions = VIEW_OPTIONS;
+	    this.textareaPlaceholder = TEXTAREA_PLACEHOLDER;
+
+	    this.state = {
+	    	questionText: "",
+	    	questionType: "Code",
+	    	questionStyle: null,
+	    	answerText: "",
+	    	answerType: "Code",
+	    	answerStyle: null,
+	    	questionArray: [],
+	    	answerArray: []
+	    }
+	}
+
+	setQuestionState(e) {
+		this.setState({ questionText: e.target.value });
+	}
+
+	setAnswerState(e) {
+		this.setState({ answerText: e.target.value });
+	}
+
+	updateQuestionType(value) {
+		this.setState({ questionType: value })
+	}
+
+	addQuestionLine() {
+		let questionType = this.state.questionType === "Code" ? "pre" : "div";
+		let newQuestion = {
+			tag: questionType,
+			class: questionType === "div" ? "text-question" : "code-question",
+			content: (this.state.questionText)
+		};
+
+		this.setState({ questionArray: this.state.questionArray.concat([newQuestion])})
+		this.setState({ questionText: "" })
+		document.getElementsByName("question-textarea").value = "";
```

Comment:
generally we avoid calling the `window` or `document` as best we can. can set the textarea valjue in the state instead?